Libraries/LibLine/Editor.cpp

```diff
@@ -976,14 +982,7 @@ void Editor::refresh_display()
     if (m_was_resized) {
         auto previous_num_columns = m_num_columns;
 
-        struct winsize ws;
-        if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &ws) < 0) {
-            m_num_columns = 80;
-            m_num_lines = 25;
-        } else {
-            m_num_columns = ws.ws_col;
-            m_num_lines = ws.ws_row;
-        }
+        get_terminal_size();
```

Comment:
Wait, why isn't it enough to do this on `SIGWINCH`?

Reply:
because we don't get that signal when we're not the owning process.
when running `cat`, the signals are delivered to `cat`, not the shell.
(we already do this on `SIGWINCH`)

Reply:
Hmm, that doesn't seem right! The kernel should `SIGWINCH` all processes on that TTY, I would think, no?

Reply:
That sounds logical, yeah.
I'll take a look.

Reply:
Nope, looks like it sends the stuff to the owning process group
why are we setting that to the child at all?

Reply:
so here's what I see:
- shell `tcsetpgrp(0, child_pgrp)`'s to avoid getting the interrupts and all that jazz
- that includes `SIGWINCH`
- shell doesn't get notified about the tty size change

if we _do_ include the shell, we have to deal with a bunch of unrelated signals, which I guess we can do...?